File: python/python-leetcode/two_sum.py
# https://leetcode.com/problems/two-sum/
# https://leetcode.com/submissions/detail/814283724/
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dictionary = {}
        for i in range(len(nums)):
            if nums[i] in dictionary:
                return [dictionary[nums[i]], i]
            else:
                dictionary[target - nums[i]] = i


sol = Solution()
print(sol.twoSum([3, 3], 6))

File: go/go-leetcode/examples/two_sum.go
package main

import "fmt"

//https://leetcode.com/problems/two-sum/

func main() {
	fmt.Println(twoSum)
}

func twoSum(nums []int, target int) []int {
	var loopA = nums
	var loopB = nums
	var myslice []int
	for i := range loopA {
		intA := loopA[i]
		for index := range loopB {
			intB := loopB[index]
			if intA+intB == target {
				if len(myslice) < 2 {
					myslice = append(myslice, i)
					myslice = append(myslice, index)
				}
			}
		}
	}
	return myslice
}
